src/mongo/db/catalog/index_catalog_impl.cpp

```diff
@@ -813,8 +813,9 @@ Status IndexCatalogImpl::_doesSpecConflictWithExisting(OperationContext* opCtx,
                                             << " already exists with different options: "
                                             << desc->infoObj());
 
-            return Status(ErrorCodes::IndexAlreadyExists,
-                          str::stream() << "index already exists with different name: " << name);
+            return Status(ErrorCodes::CannotCreateIndex,
```

Comment:
I'm curious, and couldn't quite figure it out myself - do you how exactly the IndexAlreadyExists code was being swallowed and ignored?

Reply:
I believe it is because of this block of code in index_builder.cpp (line 150-160):

 try {
                status = indexer.init(_index).getStatus();
                if (status == ErrorCodes::IndexAlreadyExists ||
                    (status == ErrorCodes::IndexOptionsConflict && _relaxConstraints)) {
                    LOG(1) << "Ignoring indexing error: " << redact(status);
                    if (allowBackgroundBuilding) {
                        // Must set this in case anyone is waiting for this build.
                        _setBgIndexStarting();
                    }
                    return Status::OK();
                }

Reply:
Oh, that totally makes sense. Thanks.

As discussed, this looks good to go, so I'll run it through the tests and see if everything passes.